src/hdmf_zarr/backend.py

```diff
@@ -566,8 +589,12 @@ def __get_ref(self, ref_object):
         source = (builder.source
                   if (builder.source is not None and os.path.isdir(builder.source))
                   else self.source)
+
         # Make the source relative to the current file
         source = os.path.relpath(os.path.abspath(source), start=self.abspath)
+
+        if export_source is not None:
+            source = '.'
```

Comment:
For future reference, I would add a TODO item to clarify what needs to be done here in the future. I think all we really need is a way to figure out whether the target for the reference will be copied into the new file on export or whether it will remain only in the source we are exporting from. Otherwise, resolution of external links is I believe already being handled. 

```suggestion
        # TODO: This check assumes that all links are internal links on export. Need to deal with external links on export.
        if export_source is not None:
            source = '.'
```